### PR TITLE
Anj/docldlt

### PR DESCRIPTION
**Pull request by andreasnoack**
The `ldltfact` documentation was wrong. The other changes ended up here after a rebase so I added them with a new commit. Are they okay to include?

### CODE REVIEW

File: base/linalg/ldlt.jl [hunk 1]
```diff
@@ -25,6 +25,12 @@ function ldltfact!{T<:Real}(S::SymTridiagonal{T})
     end
     return LDLt{T,SymTridiagonal{T}}(S)
 end
+
+doc"""
+    ldltfact(A) -> F
+
+Compute an `LDLt` factorization `F` of a `SymTridiagonal` matrix `A` such that `A=L*Diagonal(d)*L'` where `L` is a unit lower triangular matrix and `d` is a vector. The main use of an `LDLt` factorization `F=ldltfact(A)` is to solving the linear system of equations `Ax=b` with `F\b`.
```
Comment (yuyichao) [inline base/linalg/ldlt.jl]: Would be better to be consistent on whether `LDLt` is quoted (see the other doc string in `helpdb.jl`)

File: doc/stdlib/arrays.rst [hunk 2]
```diff
@@ -891,7 +891,7 @@ Sparse matrices support much of the same set of operations as dense matrices. Th
 
    .. Docstring generated from Julia source
 
-   Create a sparse matrix ``S`` of dimensions ``m x n`` such that ``S[I[k], J[k]] = V[k]``\ . The ``combine`` function is used to combine duplicates. If ``m`` and ``n`` are not specified, they are set to ``maximum(I)`` and ``maximum(J)`` respectively. If the ``combine`` function is not supplied, duplicates are added by default. All elements of ``I`` must satisfy ``1 <= I[k] <= m``\ , and all elements of ``J`` must satisfy ``1 <= J[k] <= n``\ .
+   Create a sparse matrix ``S`` of dimensions ``m x n`` such that ``S[I[k], J[k]] = V[k]``\ . The ``combine`` function is used to combine duplicates. If ``m`` and ``n`` are not specified, they are set to ``max(I)`` and ``max(J)`` respectively. If the ``combine`` function is not supplied, duplicates are added by default.
```
Comment (yuyichao) [inline doc/stdlib/arrays.rst]: Are these pending genstdlib sync?

Comment (andreasnoack) [inline doc/stdlib/arrays.rst]: I think so. When is that supposed to happen? (Sorry, I'm really new to the doc system.)

**Comment by jakebolewski** [comment]
RTFM (well Contributing.md)

File: base/linalg/ldlt.jl (hunk 1 above)
Comment (andreasnoack) [inline base/linalg/ldlt.jl]: Thanks. I'll fix that.

File: doc/stdlib/arrays.rst (hunk 2 above)
Comment (tkelman) [inline doc/stdlib/arrays.rst]: Actually this is undoing changes I made.

File: base/docs/helpdb.jl [hunk 3]
```diff
@@ -2587,13 +2587,6 @@ Right division operator: multiplication of `x` by the inverse of `y` on the righ
 Base.(:(/))
 
 doc"""
-    ldltfact(A) -> LDLtFactorization
-
-Compute a factorization of a positive definite matrix `A` such that `A=L*Diagonal(d)*L'` where `L` is a unit lower triangular matrix and `d` is a vector with non-negative elements.
```
Comment (tkelman) [inline base/docs/helpdb.jl]: Why are you removing this? Isn't there an `ldltfact` for dense matrices? The docstring that you moved inline was specifically for `SymTridiagonal`.

File: doc/stdlib/arrays.rst (hunk 2 above)
Comment (jiahao) [inline doc/stdlib/arrays.rst]: Don't delete that last sentence. Ref: #13008

Comment (tkelman) [inline doc/stdlib/arrays.rst]: Or change `max` to `maximum` - need to revert all the changes to RST except in `doc/stdlib/linalg.rst` here, something was out of sync. Probably needed to re-run `make` before running `genstdlib.jl`.

**Comment by andreasnoack** [comment]
Hold on with the comments. I messed up with the build sequence for the docs. Maybe, when you wake up tomorrow, I've managed to contribute to the documentation without breaking anything.

**Comment by andreasnoack** [comment]
I think, I had more luck this time so please review. I discovered #13068 while making this change so the two different doc strings won't work before #13068 is fixed, but the html documentation should work.

**Comment by tkelman** [comment]
lgtm. So is there no diagonal-D (modified Cholesky) `ldltfact` for dense matrices in lapack? Only indefinite `bkfact`?

**Comment by andreasnoack** [comment]
Yes. That is correct.

**Comment by tkelman** [comment]
backport candidate here?

**Comment by jakebolewski** [comment]
Yes

**Comment by tkelman** [comment]
backported in #13072

**Pull request merged**